Cycle_Sort/Cycle_Sort.java

```diff
@@ -0,0 +1,204 @@
+            for (i = start + 1; i<n; i++)  
+                if (a[i] <element)  
+                    pos += 1;  
+            while (element == a[pos])  
+                pos += 1; 
+            // put the element to it's right position     
+            if (element != a[pos]) {  
+                temp = element;  
+                element = a[pos];  
+                a[pos] = temp;    
+                writes++;  
+                }  
+            }  
+        } 
+            element = a[start];  
+            pos = start; 
+        // Find position where we put the element. We basically 
+        // count all smaller elements on right side of element.  
+        for (i = start + 1; i<n; i++)  
+            if (a[i] <element)  
+                pos++;  
+        // If element is already in correct position     
+        if (pos == start)  
+            continue; 
+        // ignore all duplicate elements      
+        while (element == a[pos])  
+            pos += 1;  
+        // put the element to it's right position
+        if (pos != start) {  
+            temp = element;  
+            element = a[pos];  
+            a[pos] = temp;    
+            writes++;  
+        } 
+        // Rotate rest of the cycle  
+        while (pos != start) {  
+            pos = start; 
+            // Find position where we put the element  
+            for (i = start + 1; i<n; i++)  
+                if (a[i] <element)  
+                    pos += 1;  
+            while (element == a[pos])  
+                pos += 1; 
+            // put the element to it's right position     
+            if (element != a[pos]) {  
+                temp = element;  
+                element = a[pos];  
+                a[pos] = temp;    
+                writes++;  
+                }  
+            }  
+        } int writes = 0,start,element,pos,temp,i;
+         // Find position where we put the element.
+        for (start = 0; start <= n - 2; start++) 
+        {   int writes = 0,start,element,pos,temp,i;
+         // Find position where we put the element.
+        for (start = 0; start <= n - 2; start++) 
+        {  
+            element = a[start];  
+            pos = start; 
+        // Find position where we put the element. We basically 
+        // count all smaller elements on right side of element.  
+        for (i = start + 1; i<n; i++)  
+            if (a[i] <element)  
+                pos++;  
+        // If element is already in correct position     
+        if (pos == start)  
+            continue; 
+        // ignore all duplicate elements      
+        while (element == a[pos])  
+            pos += 1;  
+        // put the element to it's right position
+        if (pos != start) {  
+            temp = element;  
+            element = a[pos];  
+            a[pos] = temp;    
+            writes++;  
+        } 
+        // Rotate rest of the cycle  
+        while (pos != start) {  
```

Comment:
You have the same for and while loops repeated multiple times. Please correct it.